### PR TITLE
use azure profiles instead of publishsettings

### PR DESCRIPTION
**Pull request by mischief**

### CODE REVIEW

File: auth/azure.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	Environments  []AzureEnvironment  `json:"environments"`
+	Subscriptions []AzureSubscription `json:"subscriptions"`
+}
+
+func (ap *AzureProfile) AsOptions() []azure.Options {
+	var o []azure.Options
+
+	for _, sub := range ap.Subscriptions {
+		newo := azure.Options{
+			SubscriptionID:        sub.ID,
```
Comment (crawford) [inline auth/azure.go]: A lot more of these settings are going to change. Just about everything in `AzureEnvironment` will defer depending on the region.

Comment (mischief) [inline auth/azure.go]: nothing else is used in the azure api client currently, so i don't think the environments matter right now.

Comment (crawford) [inline auth/azure.go]: StorageEndpointSuffix isn't needed when uploading the blobs?

Comment (mischief) [inline auth/azure.go]: could be, it seems i'm always using `DefaultBaseURL` currently, which is `"core.windows.net"`

**Comment by mischief** [comment]
@crawford @marineam ptal

**Comment by mischief** [comment]
this should likely be reworked to just read azure profile from `~/.azure/azureProfile.json`, like the cli and like the google token stuff in `auth` package.

File: auth/azure.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
+// Copyright 2014 CoreOS, Inc.
```
Comment (marineam) [inline auth/azure.go]: This file on the other hand *can* be 2016 ;-)

File: cmd/ore/azure/azure.go [hunk 3]
```diff
@@ -30,20 +31,53 @@ var (
...
+	if len(opts) == 0 {
+		plog.Fatalf("Azure Profile %q contains no subscriptions", azureProfile)
+	}
+
+	if azureSubscription == "" {
+		opt = &opts[0]
+	} else {
+
+		for _, o := range prof.AsOptions() {
+			if o.SubscriptionName == azureSubscription {
```
Comment (mischief) [inline cmd/ore/azure/azure.go]: sounds good.

**Comment by mischief** [comment]
retest this please

**Comment by mischief** [comment]
@crawford sounds like an off-by-one, as 2M is the chunk size.

is this reproducible with `ore azure upload-blob`? they use the same code.

**Comment by crawford** [comment]
I'm pretty sure it is, but I am not certain. We had to republish everything that was originally published by this program.

**Comment by crawford** [comment]
This is a good start. There is still some work that needs to be done around the UX, but it's good enough for now.

**Pull request merged**